tutorials/fileIO/src/test/java/com/example/fileIO/ThroughputBasedApplicationTest.java

```diff
@@ -0,0 +1,124 @@
+package com.example.fileIO;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import javax.validation.ConstraintViolationException;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.datatorrent.api.LocalMode;
+
+public class ThroughputBasedApplicationTest
+{
+  private static final String baseDirName = "target/fileIO";
+  private static final String inputDirName = baseDirName + "input-dir";
+  private static final String outputDirName = baseDirName + "output-dir";
+  private static final File inputDirFile = new File(inputDirName);
+  private static final File outputDirFile = new File(outputDirName);
+
+  private static final int numFiles = 10; // number of input files
+  private static final int numLines = 10; // number of lines in each input file
+  private static final int numPartitions = 4; // number of partitions of input operator
+
+  // create nFiles files with nLines lines in each
+  private void createFiles(final int nFiles, final int nLines) throws IOException
+  {
+    for (int file = 0; file < nFiles; file++) {
+      ArrayList<String> lines = new ArrayList<>();
+      for (int line = 0; line < nLines; line++) {
+        lines.add("file " + file + ", line " + line);
+      }
+      try {
+        FileUtils.write(new File(inputDirFile, "file" + file), StringUtils.join(lines, "\n"));
+      } catch (IOException e) {
+        System.out.format("Error: Failed to create file %s%n", file);
+        e.printStackTrace();
+      }
+    }
+    System.out.format("Created %d files with %d lines in each%n", nFiles, nLines);
+  }
+
+  private void cleanup()
+  {
+    try {
+      FileUtils.deleteDirectory(inputDirFile);
+      FileUtils.deleteDirectory(outputDirFile);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  // check that the requisite number of files exist in the output directory
+  private boolean checkFilesCopied(final int nFiles)
+  {
+    return (nFiles == FileUtils.listFiles(outputDirFile, null, false).size())
+        && (new File(outputDirFile, "file0").exists());
+  }
+
+  // return Configuration with suitable properties set
+  private Configuration getConfig()
+  {
+    final Configuration result = new Configuration(false);
+    result.setInt("dt.application.ThroughputBasedFileIO.attr.CHECKPOINT_WINDOW_COUNT", 10);
```

Comment:
Could you use something similar to following:
conf.addResource(this.getClass().getResourceAsStream("/META-INF/properties.xml")) ?